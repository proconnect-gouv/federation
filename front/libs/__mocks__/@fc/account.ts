import React from 'react';

export const AccountProvider = jest.fn();

export const AccountContext = React.createContext(expect.any(Object));
