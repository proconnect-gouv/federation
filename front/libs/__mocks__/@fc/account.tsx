import React from 'react';

export const AccountProvider = jest.fn(({ children }) => (
  <div>
    <div>AccountProvider</div>
    <div>{children}</div>
  </div>
));

export const AccountContext = React.createContext(expect.any(Object));
