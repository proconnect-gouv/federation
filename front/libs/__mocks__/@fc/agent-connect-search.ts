import * as React from 'react';

export const AgentConnectSearchContext = React.createContext({
  hasSearched: false,
  payload: {
    csrfToken: 'csrf-token-mock',
    identityProviders: [],
    isLoaded: false,
    ministries: [],
    serviceProviderName: undefined,
  },
  searchResults: [],
  setSearchTerm: () => {},
});

export const useSearchResults = jest.fn(() => ({
  searchResults: [
    {
      identityProviders: [
        {
          active: true,
          display: true,
          name: 'mock-idp-name-1.1',
          uid: 'mock-idp-uid-1.1',
        },
        {
          active: true,
          display: true,
          name: 'mock-idp-name-1.2',
          uid: 'mock-idp-uid-1.2',
        },
      ],
      ministry: {
        id: 'mock-ministry-id-1',
        identityProviders: ['mock-idp-uid-1'],
        name: 'mock-ministry-name-1',
        sort: 1,
      },
    },
    {
      identityProviders: [
        {
          active: true,
          display: true,
          name: 'mock-idp-name-2.1',
          uid: 'mock-idp-uid-2.1',
        },
        {
          active: true,
          display: true,
          name: 'mock-idp-name-2.2',
          uid: 'mock-idp-uid-2.2',
        },
      ],
      ministry: {
        id: 'mock-ministry-id-2',
        identityProviders: ['mock-idp-uid-2'],
        name: 'mock-ministry-name-2',
        sort: 0,
      },
    },
  ],
  showNoResults: false,
  showResults: true,
}));

export const useOnSearch = jest.fn(() => jest.fn());
