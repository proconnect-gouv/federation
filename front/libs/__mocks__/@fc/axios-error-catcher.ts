import React from 'react';

export const AxiosErrorCatcherProvider = jest.fn();

export const AxiosErrorCatcherContext = React.createContext(expect.any(Object));
