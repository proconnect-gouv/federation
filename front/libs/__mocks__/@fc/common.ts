export const useIsMounted = jest.fn(() => () => false);

export const useApiGet = jest.fn(() => null);

export const ucfirst = jest.fn((v) => v);

export const objectToFormData = jest.fn();

export const useLocalStorage = jest.fn(() => ({
  flush: jest.fn(),
  get: jest.fn(),
  set: jest.fn(),
}));

export enum HttpStatusCode {
  UNAUTHORIZED = 401,
}
