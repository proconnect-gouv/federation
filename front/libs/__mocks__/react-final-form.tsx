const actualModule = jest.requireActual('react-final-form');

module.exports = {
  ...actualModule,
  Field: jest.fn(({ children, ...rest }) => children(rest)),
  Form: jest.fn(({ children, ...rest }) => children(rest)),
};

export {};
