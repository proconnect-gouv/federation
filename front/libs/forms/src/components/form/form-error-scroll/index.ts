export * from './form-error-scroll.component';
