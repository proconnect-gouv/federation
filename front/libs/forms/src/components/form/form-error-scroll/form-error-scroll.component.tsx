import React from 'react';
import { FormSpy } from 'react-final-form';

import { useScrollToElement } from '@fc/common';

export const FormErrorScrollComponent = React.memo(() => {
  const { scrollToElement } = useScrollToElement('.fr-message--error');

  return (
    <FormSpy
      subscription={{ modifiedSinceLastSubmit: true, submitFailed: true }}
      onChange={() => {
        scrollToElement();
      }}
    />
  );
});

FormErrorScrollComponent.displayName = 'FormErrorScrollComponent';
