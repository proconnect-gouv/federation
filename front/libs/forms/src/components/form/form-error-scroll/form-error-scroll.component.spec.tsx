import { FormSpy } from 'react-final-form';

import { useScrollToElement } from '@fc/common';
import { renderWithFinalForm } from '@fc/testing-library';

import { FormErrorScrollComponent } from './form-error-scroll.component';

describe('FormErrorScrollComponent', () => {
  const scrollToMock = jest.fn();

  beforeEach(() => {
    // given
    jest.mocked(useScrollToElement).mockImplementation(() => ({ scrollToElement: scrollToMock }));
  });

  it('should match its snapshot', () => {
    // when
    const { container } = renderWithFinalForm(<FormErrorScrollComponent />);

    // then
    expect(container).toMatchSnapshot();
  });

  it('should render FormSpy mock with a subscription prop', () => {
    // when
    renderWithFinalForm(<FormErrorScrollComponent />);
    const { subscription } = jest.mocked(FormSpy).mock.calls[0][0];

    // then
    expect(subscription).toStrictEqual({
      modifiedSinceLastSubmit: true,
      submitFailed: true,
    });
  });

  it('should render FormSpy mock with a onChange prop', () => {
    // when
    renderWithFinalForm(<FormErrorScrollComponent />);
    const { onChange } = jest.mocked(FormSpy).mock.calls[0][0];

    // then
    expect(onChange).toBeInstanceOf(Function);
  });

  it('should call useScrollToElement when the component is render with classname parameter', () => {
    // when
    renderWithFinalForm(<FormErrorScrollComponent />);

    // then
    expect(useScrollToElement).toHaveBeenCalledOnce();
    expect(useScrollToElement).toHaveBeenCalledWith('.fr-message--error');
  });

  it('should call useScrollToElement when FormSpy onChange is called', () => {
    // when
    renderWithFinalForm(<FormErrorScrollComponent />);
    const { onChange } = jest.mocked(FormSpy).mock.calls[0][0] as { onChange: () => void };
    onChange();

    // then
    expect(scrollToMock).toHaveBeenCalledOnce();
  });
});
