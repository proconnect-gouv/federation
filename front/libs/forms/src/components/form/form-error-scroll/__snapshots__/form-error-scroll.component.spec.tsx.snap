// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormErrorScrollComponent should match its snapshot 1`] = `
<div>
  <form
    data-testid="form-wrapper"
  />
</div>
`;
